exp/runtime/internal/controllers/index.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
```

Comment:
Q: usually we keep all the index in `api/v1beta1/index`, so we can benefit of func like `AddDefaultIndexes` that gets called all across the codebase. Is there a reason to have a different code organization for this index?

Reply:
I had it like that before moving it to what it is now after discussing it together with @sbueringer  :-)

The reason was: the `api/v1beta1/index` may get used by providers too, this is currently not the case for `ExtensionConfig` / runtimev1.

Reply:
So I currently see no reason to expose it in the api package but I'd be also okay with that.

Reply:
We already have `ByClusterClassName` in the index package, let's move also this back for consistency

Reply:
In this case we would move it to `exp/runtime/api/v1alpha1/index`?

Reply:
I'm really not a fan of exposing things for consistency just because we made the mistake before

Reply:
In my opinion there is absolutely no reason why anyone else apart from us should require that index

Reply:
After talking with @sbueringer let's keep it as it is and then we will discuss code organization in our internal package (this should not be in controllers)